security-csrf/src/main/java/io/micronaut/security/csrf/session/CsrfSessionPopulator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.csrf.session;
+
+import io.micronaut.security.authentication.Authentication;
+import io.micronaut.security.csrf.CsrfConfiguration;
+import io.micronaut.security.csrf.generator.CsrfTokenGenerator;
+import io.micronaut.security.session.SessionPopulator;
+import io.micronaut.session.Session;
+import jakarta.inject.Singleton;
+
+/**
+ * Populates the session with a CSRF token.
+ * @author Sergio del Amo
+ * @since 4.11.0
+ * @param <T> Request
+ */
+@Singleton
+public class CsrfSessionPopulator<T> implements SessionPopulator<T> {
```

Comment:
Non-final and public by design?

Reply:
I made it final and package private https://github.com/micronaut-projects/micronaut-security/pull/1826/commits/c70e6e6b27e893330bf0f89b7982d3c4bbc0477f